frontend/src/scenes/annotations/logic.js

```diff
@@ -65,7 +65,8 @@ export const annotationsTableLogic = kea({
             }
             actions.appendAnnotations(results)
         },
-        [annotationsModel.actions.createGlobalAnnotation]: () => {
+        [annotationsModel.actions.createGlobalAnnotation]: async ({}, breakpoint) => {
```

Comment:
So actually I'm not sure about this, there's no guarantee that the initial request will be completed in 200ms. I think the right approach would be to either refactor the creation method to a loader (https://kea.js.org/docs/guide/concepts/#listeners) [or equivalent actions/reducers/listener] where you actually listen to the success event.

Alternatively you could simply call an action after the request is successful with which you update the list.

Reply:
Agree with this. We're not sure if the global annotation has already been created after the `loadAnnotations` action fires.

The `createGlobalAnnotation` and `delete*` actions in the `annotationsModel` should probably go into the `globalAnnotations` loader. This way you'll have a `createGlobalAnnotationSuccess` action you can use from `annotations/logic.js`.

Also, definitely outside the scope of this PR, but in case you feel like some TypeScript practice, converting this small `logic.js` to `logic.ts` would be a good learning experience.

Reply:
That's super clean! Thanks for the suggestion. One thing I noticed though is that it's hard to bring `deleteGlobalAnnotation` into a loader, as we can't wait for synchronous `deleteWithUndo` to complete:

```
loaders: () => ({
   globalAnnotations: {
        ...
        deleteGlobalAnnotation: async ({ id }) => {
            if (id >= 0) {
                // Because deleteWithUndo is a synchronous callback, we can't wait for a delete to succeed. A partial failure here with this implementation would create a data mismatch.
                deleteWithUndo({
                    endpoint: 'annotation',
                    object: { name: 'Annotation', id },
                    callback: () => actions.loadGlobalAnnotations({}),
                })
            }
            return values.globalAnnotations.filter((annotation) => annotation.id !== id)
        },
        ...
   }
}}
```

Given this limitation (and seeing as we already [do that ](https://github.com/alexkim205/posthog/blob/2bee644de09dc0b0684faf999ce7fe721b086a20/frontend/src/scenes/experimentation/featureFlagLogic.tsx#L119-L127) elsewhere), I think it'll be okay to bring `createGlobalAnnotation` into a loader and keep `deleteGlobalAnnotation` as a listener.

Reply:
Definitely fine to leave as you described. There is however a way to convert sync code with callbacks to async code that you can await for with `new Promise((resolve, reject) => syncFunc({ callback: () => resolve() })` or something equivalent. I'm not sure if that's the right approach here, so use your best judgement.